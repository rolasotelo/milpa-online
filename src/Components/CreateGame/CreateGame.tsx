import React from "react";

interface Props {}

const CreateGame = (props: Props) => {
  return <div>Create Game</div>;
};

export default CreateGame;
