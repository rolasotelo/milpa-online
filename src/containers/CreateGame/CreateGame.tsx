import React from "react";
import { useTranslation } from "react-i18next";

interface Props {
  onClickCreate: () => void;

}

function PlayButton(props: { onClick: () => void; text: string }) {
  const { onClick, text } = props;
  return (
    <button
      type="button"
      className="z-10  bg-button-pink mt-4 w-52 h-24  focus:outline-none focus:bg-button-pink-pressed pl-6 pt-5 focus:pl-4 focus:pt-3 text-milpaBlue-dark"
      onClick={onClick}
      style={{
        fontFamily: "goodlife-serif, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",
        fontSize: "1.75rem",
      }}
    >
      {text}
    </button>
  );
}

function CreateGame(props: Props) {
  const { t } = useTranslation();
  const { onClickCreate} = props;
  return (
    <div className="w-20.38rem md:w-27.5rem h-32.5rem bg-create-background ring-8 ring-inset ring-milpaBlue-dark m-5">
      <div className="flex justify-center">
       <PlayButton onClick={onClickCreate} text={t("play.create.button")}/>
      </div>
    </div>
  );
}

export default CreateGame;
