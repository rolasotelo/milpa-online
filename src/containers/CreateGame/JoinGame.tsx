import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { PropsWithChildren } from "../../common/interfaces";

interface Props {
  onClickJoin: (code: string) => void;
}

function JoinButton(props: { onClick: () => void; text: string }) {
  const { onClick, text } = props;
  return (
    <button
      type="button"
      className="z-10  bg-button-pink mt-4 w-52 h-24  focus:outline-none focus:bg-button-pink-pressed pl-6 pt-5 focus:pl-4 focus:pt-3 text-milpaBlue-dark"
      onClick={onClick}
      style={{
        fontFamily: "goodlife-serif, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",
        fontSize: "1.75rem",
      }}
    >
      {text}
    </button>
  );
}

function Layout(props: PropsWithChildren) {
  const { children } = props;
  return (
    <div className="relative z-10 w-20.38rem tablet:w-27.5rem h-32.5rem bg-join-background ring-8 ring-inset ring-milpaBlue-dark m-5">
      <div className="absolute left-3 top-3 bg-symbol-corner-1 w-8 h-8" />
      <div className="absolute right-3 top-3 bg-symbol-corner-4 w-8 h-8" />
      <div className="flex flex-col">{children}</div>
    </div>
  );
}

function JoinGame(props: Props) {
  const { onClickJoin } = props;
  const [gameCode, setGameCode] = useState("");
  const { t } = useTranslation();
  const handleClick = React.useCallback(() => {
    onClickJoin(gameCode);
  }, [onClickJoin, gameCode]);
  const handleChange = React.useCallback((event) => {
    setGameCode(event.target.value);
  }, []);
  return (
    <Layout>
      <div className="mx-auto">
        <JoinButton onClick={handleClick} text={t("play.join.button")} />
      </div>
      <div className="mx-auto">
        <input
          className="z-40 placeholder-milpaBeige-default placeholder-opacity-100 h-input-3d w-input-3d text-center bg-pink-input bg-milpaBlue-default focus:outline-none text-milpaBeige-default"
          placeholder={t("play.join.paste")}
          value={gameCode}
          onChange={handleChange}
        />
      </div>
    </Layout>
  );
}

export default JoinGame;
