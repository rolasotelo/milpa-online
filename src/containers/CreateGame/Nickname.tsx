import React from "react";
import { useTranslation } from "react-i18next";

interface Props {
  nickname: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

function Nickname(props: Props) {
  const { nickname, onChange } = props;
  const { t } = useTranslation();
  return (
    <div className='overflow-x-hidden' >
    <div
      className=" w-20.38rem md:w-49.81rem h-4.38rem mx-auto flex justify-around bg-green-banner-mobile md:bg-green-banner md:px-5 pt-6 mt-2 text-2xl md:text-3xl"
      style={{
        fontFamily: "goodlife-sans-condensed, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",

      }}
    >
      <div
        className=" md:ml-6 text-milpaBeige-default"

      >
        <a href="/play">{t("play.greeting").toUpperCase()}</a>
      </div>
      <div className="md:ml-2">
        <input
          className=" placeholder-gray-500 placeholder-opacity-100 focus:outline-none bg-transparent text-milpaBlue-dark"
          placeholder="Gabinka?"
          value={nickname}
          onChange={(event) => {
            onChange(event);
          }}
        />
      </div>
    </div>
    </div>
  );
}

export default Nickname;
