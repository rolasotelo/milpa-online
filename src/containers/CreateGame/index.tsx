import React from "react";
import { hot } from "react-hot-loader/root";
import CreateGame from "./CreateGame";
import JoinGame from "./JoinGame";
import Layout from "../../components/Layout";
import Nickname from "./Nickname";
import useCreateGameContext from "../../providers/CreateGameProvider/useCreateGameContext";

function CreateOrJoinGame() {
  const context = useCreateGameContext();
  const { onClickCreate, onClickJoin} = context;
  return (
    <Layout>
      <Nickname
        nickname={context.nickname}
        onChange={context.onChangeNickname}
        greeting="Hello"
      />
      <div className="flex flex-col md:flex-row justify-evenly items-center relative">
        <CreateGame onClickCreate={onClickCreate} />
        <JoinGame onClickJoin={onClickJoin} />
        <div className="absolute inset-x-0 top-32  tablet:top-44 bg-mountains-herobox mx-auto max-w-100vw tablet:w-mountains h-mountains" />
      </div>
    </Layout>
  );
}

export default hot(CreateOrJoinGame);
