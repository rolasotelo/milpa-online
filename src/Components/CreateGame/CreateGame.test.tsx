import React from "react";
import renderer from "react-test-renderer";

import CreateGame from "./CreateGame";

describe("<CreateGame />", () => {
  const defaultProps = {
    onClickCreate: () => {},
    text: "CREATE",
  };
  const wrapper = renderer.create(<CreateGame {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
