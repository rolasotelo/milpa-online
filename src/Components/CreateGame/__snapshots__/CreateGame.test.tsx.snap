// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateGame /> render 1`] = `
<div
  className="w-20.38rem md:w-27.5rem h-32.5rem bg-create-background ring-8 ring-inset ring-black m-5"
>
  <div
    className="flex justify-center"
  >
    <button
      className="bg-button-pink w-52 h-24 mt-4 focus:outline-none focus:bg-button-pink-pressed pl-3 pb-2 focus:pl-2 focus:pb-1"
      onClick={[Function]}
      style={
        Object {
          "fontFamily": "goodlife-sans-condensed, sans-serif",
          "fontSize": "2rem",
          "fontStyle": "normal",
          "fontWeight": 400,
        }
      }
    >
      CREATE
    </button>
  </div>
</div>
`;
